src/dmd/expressionsem.d

```diff
@@ -2551,14 +2551,20 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             if (exp.fd.treq)
                 printf("  treq = %s\n", exp.fd.treq.toChars());
         }
+
+        if (exp.type)
```

Comment:
Shouldn't this be : 

```d
if (exp.type && exp.type != Type.tvoid)
```

?

Reply:
According to @9rnsr, no.  He addressed the check with this comment.

> But today, the post-semantic process is handled in `FuncExp.matchType`, and the `type == Type.tvoid` check is not used anymore. That's why the check can be removed now.

Can you think of a test case that would justify adding the check against `Type.tvoid`?

Reply:
I can't say I understand completely the reason behind the `Type.tvoid` check. Chesterton's fence...